cryostatio/cryostat-operator
# chore(grafana): remove separate Grafana service/route/ingress

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Depends on #808
Based on #808
Related to #710

## Description of the change:
Grafana should now only be accessible through the auth proxy at the `/grafana` path. It should not have its own separate service, or Route/Ingress, because all traffic must flow through the auth proxy now.

## Motivation for the change:
This ensures that Grafana is only accessible using the same authn/authz as the rest of the application.

## How to manually test:
1. Check out PR, build with `OPENSHIFT_OAUTH_PROXY_IMG=quay.io/andrewazores/openshift-oauth-proxy:test-14`. Deploy Operator and create a Cryostat CR with `enableCertManager: false`
2. Open Cryostat web UI and log in via OpenShift SSO as cluster admin
3. Create a `localhost:0` custom target
4. Start a recording, give it a few moments, then "View in Grafana" it. This should open a new browser tab. The URL should have the same `https` scheme and the same  domain, but should differ in the path - the Grafana dashboard should be at `https://cryostat-example.apps.crc-testing/grafana`. You should not need to re-login to view this.
5. Copy the Grafana dashboard URL, then paste into a new private/incongnito window and go. You should be prompted to log in via OpenShift SSO. Log in as cluster admin and you should see the same data as in the previous step. Remove `/grafana` from the URL and you should go back to the Cryostat UI, without needing to re-login.
6. Archive the recording, then repeat steps 4 and 5 on that.
7. `oc get routes` and `oc get services` and verify that there is only one Service/Route associated with the Cryostat instance, which points to port `4180` - the authproxy HTTP port.
